### PR TITLE
Fixes #12019 - add errata global status

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

File: app/models/katello/concerns/content_aspect_host_extensions.rb [hunk 1]
```diff
@@ -4,11 +4,23 @@ module ContentAspectHostExtensions
       extend ActiveSupport::Concern
 
       included do
+       ERRATA_STATUS_MAP = {
```
Comment (ehelms) [inline app/models/katello/concerns/content_aspect_host_extensions.rb]: I think this is off by a space

File: app/models/katello/errata_status.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      else
+        ::HostStatus::Global::WARN
+      end
+    end
+
+    def to_status(_options = {})
+      errata = host.content_aspect.try(:applicable_errata)
+      if host.content_aspect.nil?
+        UNKNOWN
+      elsif errata.where("#{Katello::Erratum.table_name}.errata_type" => Erratum::SECURITY).any?
```
Comment (ehelms) [inline app/models/katello/errata_status.rb]: Could use the `security` scope here?

File: app/models/katello/errata_status.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        NEEDED_ERRATA
+      elsif host.content_aspect.bound_repositories.empty?
+        UNKNOWN
+      else
+        UP_TO_DATE
+      end
+    end
+
+    def relevant?
+      host.content_aspect
```
Comment (ehelms) [inline app/models/katello/errata_status.rb]: If `host.content_aspect` returns `nil` then the errata status won't be shown, is that correct? Do you then need the check on line 44?

File: app/models/katello/errata_status.rb [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        ::HostStatus::Global::OK
+      when UNKNOWN
+        ::HostStatus::Global::WARN
+      else
+        ::HostStatus::Global::WARN
+      end
+    end
+
+    def to_status(_options = {})
+      errata = host.content_aspect.try(:applicable_errata)
```
Comment (ehelms) [inline app/models/katello/errata_status.rb]: Do you want a similar early check here as https://github.com/Katello/katello/pull/5514/files#diff-eff3ce936eaeca07c83b52b05e8dee47R37 ?

File: app/models/katello/errata_status.rb (hunk 3 above)
Comment (jlsherrill) [inline app/models/katello/errata_status.rb]: it won't, but i was attempting to handle the situation where to_status is called without a relevant check.  It shouldn't happen but I'd prefer to handle the case than cause an exception.

File: app/models/katello/errata_status.rb (hunk 4 above)
Comment (jlsherrill) [inline app/models/katello/errata_status.rb]: so there is a similar check below in the very next line.  In this case it made more much sense to include it in the if statement i think

Comment (jlsherrill) [inline app/models/katello/errata_status.rb]: which wasn't feasible in https://github.com/Katello/katello/pull/5514/files#diff-eff3ce936eaeca07c83b52b05e8dee47R37

Comment (ehelms) [inline app/models/katello/errata_status.rb]: I'd say then just do like you did there and avoid an unneeded call to get the errata given the content_aspect being nil is unrelated to the rest of if block which revolves around the errata.

Comment (jlsherrill) [inline app/models/katello/errata_status.rb]: well the errata call is not really doing anything as the AR query is lazy loaded, but i can make that change

**Comment by jlsherrill** [comment]
saw an issue with more than 1 errata status for a host, will investigate

**Comment by ehelms** [comment]
@jlsherrill ping

**Comment by jlsherrill** [comment]
@ehelms this is waiting on host_unification to go to master

**Comment by ehelms** [comment]
So this will be closed and re-opened against master once the host_unification branch goes in?

**Comment by jlsherrill** [comment]
Yes, i can go ahead and rebase (and probably will), but it won't in the end
go into the host_unification branch

**Comment by jlsherrill** [comment]
Closing in favor of https://github.com/Katello/katello/pull/5667

**Pull request closed without merging**